recipes/twitch-native-ipc/all/conanfile.py

```diff
@@ -0,0 +1,72 @@
+from conans import ConanFile, CMake, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class TwitchNativeIpcConan(ConanFile):
+    name = "twitch-native-ipc"
+    license = "MIT"
+    homepage = "https://github.com/twitchtv/twitch-native-ipc"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "Twitch natve ipc library"
+    topics = ("<twitch>", "<ipc>")
+    settings = "os", "compiler", "build_type", "arch"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True, "libuv:shared": False}
+    generators = "cmake"
+    exports = ["CMakeLists.txt", "patches/**"]
+    requires = "libuv/1.38.1"
+
+    _cmake = None
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.settings.os != "Windows" and self.settings.os != "Macos":
```

Comment:
Why did you exclude Linux?
The section [*Running tests*](https://github.com/twitchtv/twitch-native-ipc#running-tests) mentions Unix, so I assume Linux is supported, or isn't it?

Reply:
Unix was mentioned more for Macos than for Linux.  The current code as is will fail to compile on linux.

Reply:
I can fix it on my Linux system using the following patch
```patch
diff --git a/libnativeipc/CMakeLists.txt b/libnativeipc/CMakeLists.txt
index fe5cc30..e8f20cd 100644
--- a/libnativeipc/CMakeLists.txt
+++ b/libnativeipc/CMakeLists.txt
@@ -22,7 +22,7 @@ target_sources(nativeipc PRIVATE
   src/LogMacrosWithHandle.h
   src/Message.h
   src/OperationQueue.cpp
-  src/OperationQUeue.h
+  src/OperationQueue.h
   src/Pipe-ClientTransport.cpp
   src/Pipe-ClientTransport.h
   src/Pipe-ServerTransport.cpp
diff --git a/libnativeipc/include/nativeipc/ConnectionFactory.h b/libnativeipc/include/nativeipc/ConnectionFactory.h
index 0dd3aaa..3d6cc8c 100644
--- a/libnativeipc/include/nativeipc/ConnectionFactory.h
+++ b/libnativeipc/include/nativeipc/ConnectionFactory.h
@@ -2,8 +2,8 @@
 
 #include "ConnectionExports.h"
 #include "IServerConnection.h"
+#include <memory>
 #include <string>
-
 namespace Twitch::IPC::ConnectionFactory {
 NATIVEIPC_LIBSPEC std::unique_ptr<IConnection> newServerConnection(const std::string &endpoint, bool allowMultiuserAccess = false);
 NATIVEIPC_LIBSPEC std::unique_ptr<IConnection> newClientConnection(const std::string &endpoint);
diff --git a/libnativeipc/include/nativeipc/IConnection.h b/libnativeipc/include/nativeipc/IConnection.h
index 30e4db3..9224c98 100644
--- a/libnativeipc/include/nativeipc/IConnection.h
+++ b/libnativeipc/include/nativeipc/IConnection.h
@@ -1,5 +1,5 @@
 #pragma once
-
+#include <cstring>
 #include <functional>
 #include <string>
 #include <vector>
diff --git a/libnativeipc/src/ConnectionBase.h b/libnativeipc/src/ConnectionBase.h
index a5e9371..d0711b3 100644
--- a/libnativeipc/src/ConnectionBase.h
+++ b/libnativeipc/src/ConnectionBase.h
@@ -4,7 +4,7 @@
 #include "IConnection.h"
 #include <atomic>
 #include <mutex>
-
+#include <memory>
 namespace Twitch::IPC {
 class IClientTransport;
 class IServerTransport;
diff --git a/libnativeipc/src/ConnectionFactoryPrivate.h b/libnativeipc/src/ConnectionFactoryPrivate.h
index 9e4aa64..ccf5f50 100644
--- a/libnativeipc/src/ConnectionFactoryPrivate.h
+++ b/libnativeipc/src/ConnectionFactoryPrivate.h
@@ -2,7 +2,7 @@
 
 #include "IClientTransport.h"
 #include "IServerTransport.h"
-
+#include <memory>
 namespace Twitch::IPC::ConnectionFactory {
 
 // ReSharper disable once CppPolymorphicClassWithNonVirtualPublicDestructor
diff --git a/libnativeipc/src/OperationQueue.h b/libnativeipc/src/OperationQueue.h
index 5778382..abb3243 100644
--- a/libnativeipc/src/OperationQueue.h
+++ b/libnativeipc/src/OperationQueue.h
@@ -1,6 +1,7 @@
 #pragma once
 
 #include "DeleteConstructors.h"
+#include <condition_variable>
 #include <functional>
 #include <mutex>
 #include <queue>
```

Reply:
Your + this patch would be useful for upstream too, I think.